huaweicloud/services/iotda/resource_huaweicloud_iotda_device_async_command.go

```diff
@@ -0,0 +1,173 @@
+package iotda
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/iotda/v5/model"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API IoTDA POST /v5/iot/{project_id}/devices/{device_id}/async-commands
+// @API IoTDA GET /v5/iot/{project_id}/devices/{device_id}/async-commands/{command_id}
+func ResourceDeviceAsyncCommand() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDeviceAsyncCommandCreate,
+		ReadContext:   resourceDeviceAsyncCommandRead,
+		DeleteContext: resourceDeviceAsyncCommandDelete,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"device_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"send_strategy": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"service_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"paras": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"expire_time": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"result": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"sent_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"delivered_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"response_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildDeviceAsyncCommandBodyParams(d *schema.ResourceData) *model.CreateAsyncCommandRequest {
+	commandParas := utils.ValueIgnoreEmpty(d.Get("paras"))
```

Comment:
The response value of function `utils.ValueIgnoreEmpty(d.Get("paras"))` may be nil.
The logic on line `102` may result in a null pointer.